chajda/tsquery/augments.py

```diff
@@ -53,114 +58,225 @@ def augments_gensim(lang, word, config=Config(), n=5):
+# NOTE:
+# this dictionary is necessary for using annoy for nearest neighbor query after removing fasttext download
+fasttext_index_for_word = {}
 
-def create_annoy_index_if_needed(lang, word='google', n=5, seed=22):
+# fasttext_word_at_index dictionary will contain values for all words in a fasttext language dictionary,
+# and keys corresponding to the index of a word;
+# it is populated by the load_annoy_index function,
+# and subsequently written to a json file
+# NOTE:
+# this dictionary is necessary for using annoy for nearest neighbor query after removing fasttext download
+fasttext_word_at_index = {}
+
+
+def load_annoy(lang, word='google', n=5, seed=22, run_destroy_fasttext = True):
     '''
     Checks whether an annoy index has already been created for a specified language.
     Populates an annoy index  with vectors from the fasttext model that corresponds to language,
-    builds the index, saves it to disk. 
-    The annoy index is stored in the annoy_indices dictionary.
-
+    builds the index, saves it to disk.
+    The annoy index is stored in the annoy dictionary.
     lang: language passed from augments_fasttext
-    
+
     NOTE:
     Populating an annoy index with vectors from a fasttext model requires more memory than github actions allows for.
     Therefore, the doctest has been commented out.
-
     # >>> create_annoy_index_if_needed('en') or annoy_indices['en'] != None
     # True
     '''
     # create annoy index if not already created
     try:
         annoy_indices[lang]
     except KeyError:
-        annoy_indices[lang] = AnnoyIndex(300, 'angular')
+        annoy_indices[lang] = namedtuple(f"annoy", ['index', 'json1', 'json2'])
+        annoy_indices[lang].index = AnnoyIndex(300, 'angular')
 
         # if annoy index has not been saved for this language yet,
         # populate annoy index with vectors from corresponding fasttext model
         try:
-            annoy_indices[lang].load(f"{fasttext_annoy_data_path}/annoy/{lang}{seed}.ann")
+            annoy_indices[lang].index.load(f"{fasttext_annoy_data_path}/annoy/{lang}{seed}.ann")
 
         # OSError occurs if index is not already saved to disk
         except OSError:
+            # download fasttext model if not already downloaded
+            try:
+                fasttext_models[lang]
+            except KeyError:
+                load_fasttext(lang)
+
             # The annoy library uses a random number genertor when building up the trees for an Annoy Index.
             # In order to ensure that the output is deterministic, we specify the seed value for the random number generator.
-            annoy_indices[lang].set_seed(seed)
+            annoy_indices[lang].index.set_seed(seed)
 
             # Populate annoy index with vectors from corresponding fasttext model
             i = 0
-            for j in fasttext_models[lang].words:
+            for j in fasttext_models[lang].get_words(on_unicode_error='replace'):
```

Comment:
did this 100% fix the problem you were having before?  If so, you should leave a comment in the github issue you found explaining how to fix it for future people who find it